syntax = "proto3";

package TKFilter;

import "google/protobuf/empty.proto";

service TKFilterService {

    // Setup Emmt
    rpc SetupEmmt (stream SetupEmmtRequest) returns (google.protobuf.Empty) {}

    // Setup Xset
    rpc SetupXset (stream SetupXsetRequest) returns (google.protobuf.Empty) {}

    // Search
    rpc Search (SearchRequest) returns (SearchResponse) {}
}

message c_ectr
{
    bytes c = 1;
    bytes ectr = 2;
}

message SetupEmmtRequest
{
    bytes key = 1;
    repeated c_ectr value = 2;
}

message SetupXsetRequest
{
    bytes xset_v = 1;
}

message SearchRequest
{
    bytes tokp = 1;
    message cks {
        repeated int32 ck = 1;
    }
    repeated cks ck = 2;
}

message SearchResponse {
    repeated c_ectr ct = 1;
}